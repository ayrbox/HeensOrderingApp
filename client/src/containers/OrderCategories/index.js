import { connect } from 'react-redux';
import Categories from './OrderCategories';

import { setCategory } from '../../actions/newOrderActions';

const mapState = state => ({
  categoryId: state.newOrder.categoryId,
});

export default connect(mapState, {
  setCategory,
})(Categories);
