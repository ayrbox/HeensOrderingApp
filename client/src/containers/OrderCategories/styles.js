const styles = theme => ({
  content: {
    padding: `${theme.spacing.unit}px`,
  },
  categoryButton: {
    width: '100%',
    marginBottom: '10px',
  },
  categorySelectionIcon: {
    marginRight: 0,
  },
});

export default styles;
