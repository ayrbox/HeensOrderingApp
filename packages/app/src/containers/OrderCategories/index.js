import { connect } from 'react-redux';
import Categories from './OrderCategories';

import { setCategory } from '../../store/newOrder';

const mapState = state => ({
  categoryId: state.newOrder.categoryId,
});

export default connect(mapState, {
  setCategory,
})(Categories);
