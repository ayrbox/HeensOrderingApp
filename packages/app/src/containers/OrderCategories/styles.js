const styles = {
  categorySelectionIcon: {
    marginRight: 0,
  },
};

export default styles;
