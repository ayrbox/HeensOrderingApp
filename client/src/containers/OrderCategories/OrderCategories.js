import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import InboxIcon from '@material-ui/icons/KeyboardArrowRight';
import Fetch from '../../components/Fetch';


import styles from './styles';

const Categories = ({
  classes,
  category,
  setCategory,
}) => (
  <List>
    <Fetch url="/api/categories/">
      {({ data, loading }) => {
        if (loading) {
          return 'Loading....';
        }
        return [{ _id: '', name: 'All' }, ...data].map(({
          _id: id,
          name,
        }) => (
          <ListItem
            key={id}
            button
            aria-label="category"
            onClick={(e) => {
              e.preventDefault();
              setCategory(id);
            }}
          >
            <ListItemText>{name}</ListItemText>
            {category === id && (
              <ListItemIcon className={classes.categorySelectionIcon}>
                <InboxIcon />
              </ListItemIcon>
            )}
          </ListItem>
        ));
      }}
    </Fetch>
  </List>
);

Categories.defaultProps = {
  category: '',
};

Categories.propTypes = {
  classes: PropTypes.shape().isRequired,
  category: PropTypes.string,
  setCategory: PropTypes.func.isRequired,
};

export default withStyles(styles)(Categories);
