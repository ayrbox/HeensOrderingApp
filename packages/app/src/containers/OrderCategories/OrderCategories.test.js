import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import mockAxios from 'axios';
import '@testing-library/jest-dom/extend-expect';

import OrderCategories from './OrderCategories';
import { waitForElement } from '@testing-library/dom';

const sampleCategories = [{
  _id: '5b3dfd4116987b4e407f8652',
  name: 'Appetisers',
  description: 'Appetisers Menu',
}, {
  _id: '5b58ae049841684dcc644f43',
  name: 'Black Bean',
  description: 'Black Bean',
}, {
  _id: '5b58adf99841684dcc644f42',
  name: 'Beef',
  description: 'Beef Items',
}, {
  _id: '5b58ae0b9841684dcc644f44',
  name: 'Chicken',
  description: 'Chicken',
}, {
  _id: '5b58ae109841684dcc644f45',
  name: 'Curry',
  description: 'Curry',
}, {
  _id: '5b58ae1e9841684dcc644f47',
  name: 'Drinks',
  description: 'Drinks',
}, {
  _id: '5b58ae239841684dcc644f48',
  name: 'Duck',
  description: 'Duck',
}, {
  _id: '5b58ae169841684dcc644f46',
  name: 'Desert',
  description: 'Desert',
}];

jest.mock('axios');
const mockSetCategory = jest.fn();

describe('<OrderCategories />', () => {
  it('should render correctly', async () => {
    const testData = {
      data: sampleCategories,
    };

    mockAxios.get.mockResolvedValueOnce(testData);
    const { container, getByTestId, getAllByTestId } = render(
      <OrderCategories
        setCategory={mockSetCategory}
      />,
    );
    expect(getByTestId(/category-list/)).toHaveTextContent(/Loading/);

    const list = await waitForElement(() => getAllByTestId(/category-item/));

    expect(list.length).toEqual(sampleCategories.length + 1); // + All (default)
    list.forEach((item) => {
      fireEvent.click(item);
    });

    expect(mockSetCategory).toHaveBeenCalledTimes(sampleCategories.length + 1); // + All (default);
    expect(container).toMatchSnapshot();
  });
});
