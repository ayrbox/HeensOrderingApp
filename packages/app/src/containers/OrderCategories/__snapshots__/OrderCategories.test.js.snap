// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderCategories /> should render correctly 1`] = `
<div>
  <ul
    class="MuiList-root-2 MuiList-padding-3"
    data-testid="category-list"
  >
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          All
        </span>
      </div>
      <div
        class="MuiListItemIcon-root-63 Categories-categorySelectionIcon-1"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-64"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Appetisers
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Black Bean
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Beef
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Chicken
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Curry
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Drinks
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Duck
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
    <div
      aria-label="category"
      class="MuiButtonBase-root-18 MuiListItem-root-6 MuiListItem-default-9 MuiListItem-gutters-14 MuiListItem-button-15"
      data-testid="category-item"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemText-root-21"
      >
        <span
          class="MuiTypography-root-27 MuiTypography-subheading-34 MuiListItemText-primary-24"
        >
          Desert
        </span>
      </div>
      <span
        class="MuiTouchRipple-root-73"
      />
    </div>
  </ul>
</div>
`;
